include/cm/nccl_ofi_cm_resources.h

```diff
@@ -29,7 +30,18 @@ class endpoint
 public:
 	/* Memory registration handle */
 	struct mr_handle_t {
-		fid_mr *mr;
+		/* Default constructor */
+		mr_handle_t() = default;
+
+		/* Move constructor and assignment */
+		mr_handle_t(mr_handle_t&&) = default;
+		mr_handle_t& operator=(mr_handle_t&&) = default;
+
+		/* Delete copy operations since smart pointers are non-copyable */
+		mr_handle_t(const mr_handle_t&) = delete;
+		mr_handle_t& operator=(const mr_handle_t&) = delete;
```

Comment:
Note: mentioned in the PR description, but I updated classes that have the unique_ptr Libfabric resources and are used as non-pointer members to be not copy-able to meet the `unique_ptr` requirement (e.g. this and CM `endpoint ` class, the RDMA protocol rail structs that are stored in STL vectors). I did not yet update the classes that are only used as pointers (e.g. the main transport datatypes with unique_ptr Libfabric resources like device/domain/endpoint) to not be copy-able.